optimum/intel/openvino/modeling_decoder.py

```diff
@@ -354,6 +356,18 @@ class OVModelForCausalLM(OVBaseDecoderModel, GenerationMixin):
     export_feature = "text-generation"
     auto_model_class = AutoModelForCausalLM
 
+    def generate(self, *args, **kwargs):
+        self.compile()
+        infer_context = [self.compiled_model.create_infer_request()]
+        kwargs["infer_context"] = infer_context
+        return super().generate(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
```

Comment:
@dtrawins, please explain why `__call__` should be different from `forward` behavior. `__call__` will eventually call `forward` without any added semantics on top of that. So why cannot we move this code to forward?

Reply:
@slyalin that is indeed not required. potentially it could be used to pass the infer_request context. Assuming we create new request in the forward method if generate method didn't pass the context, that is not required.